pallets/handles/src/handles_signed_extension.rs

```diff
@@ -80,34 +85,41 @@ pub fn map_dispatch_error(err: DispatchError) -> InvalidTransaction {
 	})
 }
 
-#[allow(deprecated)]
-impl<T: Config + Send + Sync> SignedExtension for HandlesSignedExtension<T>
+/// The info passed between the validate and prepare steps for the `HandlesSignedExtension` extension.
+#[derive(RuntimeDebugNoBound)]
+pub enum Val {
+	/// Valid transaction, no weight refund.
+	Valid,
+	/// Weight refund for the transaction.
+	Refund(Weight),
+}
+
+/// The info passed between the prepare and post-dispatch steps for the `HandlesSignedExtension` extension.
+#[derive(RuntimeDebugNoBound)]
+pub enum Pre {
+	/// Valid transaction, no weight refund.
+	Valid,
+	/// Weight refund for the transaction.
+	Refund(Weight),
+}
+
+impl<T: Config + Send + Sync> TransactionExtension<T::RuntimeCall> for HandlesSignedExtension<T>
 where
 	T::RuntimeCall: Dispatchable<Info = DispatchInfo> + IsSubType<Call<T>>,
+	<T as frame_system::Config>::RuntimeOrigin: AsSystemOriginSigner<T::AccountId> + Clone,
 {
-	type AccountId = T::AccountId;
-	type Call = T::RuntimeCall;
-	type AdditionalSigned = ();
-	type Pre = ();
 	const IDENTIFIER: &'static str = "HandlesSignedExtension";
+	type Implicit = ();
+	type Val = Val;
+	type Pre = Pre;
 
-	/// Additional signed
-	fn additional_signed(&self) -> core::result::Result<(), TransactionValidityError> {
-		Ok(())
-	}
-
-	/// Pre dispatch
-	#[allow(deprecated)]
-	fn pre_dispatch(
-		self,
-		who: &Self::AccountId,
-		call: &Self::Call,
-		info: &DispatchInfoOf<Self::Call>,
-		len: usize,
-	) -> Result<Self::Pre, TransactionValidityError> {
-		self.validate(who, call, info, len).map(|_| ())
+	fn weight(&self, call: &T::RuntimeCall) -> Weight {
+		if let Some(Call::retire_handle {}) = call.is_sub_type() {
+			T::WeightInfo::validate_retire_handle_benchmark()
+		} else {
+			Weight::zero()
```

Comment:
If a transaction is going to take the weighted path charge it upfront, for no-op type tx passthrough with zero weight